.github/workflows/build-and-deploy.yaml

```diff
@@ -0,0 +1,56 @@
+name: build and deploy mtb
+
+on:
+  workflow_call:
+    inputs:
+      branch:
+        required: true
+        type: string
+        description: The branch to checkout
+      destination_bucket:
+        required: true
+        type: string
+        description: The bucket to deploy to
+      distribution_id:
+        required: true
+        type: string
+        description: The distribution to invalidate
+
+jobs:
+  build:
+    name: build-deploy
+    runs-on: ubuntu-latest
+    # Allow OIDC Integration (so we can assume the AWS role to deploy)
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3.0.0
+        with:
+          ref: ${{ inputs.branch }}
+      - name: Use Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+          cache: yarn
+      - name: build
+        run: |
+          export CI=false
+          yarn install --check-files
+          yarn build
+      - name: Assume AWS Role
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::797640923903:role/sagebase-github-oidc-sage-bionetworks-mtb-sageit
+          role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+          role-duration-seconds: 1200
```

Comment:
I'm not very familiar with our AWS roles configuration -- can the role be used when running the workflow from any repo (e.g. a forked repo) or is the role scoped to the `Sage-Bionetworks/mtb` repo? 

If the repo owner and repo name were updated in the future (e.g. `Sage-Bionetworks/mtb` -> `BridgeDigitalHealth/open-bridge`), would the role need to be updated?

Reply:
Also, the `role-to-assume` includes "mtb" - is that tied to the Mobile Toolbox Researcher UI in any way? Can we change that as well.

Reply:
The role is tied to Sage-Bionetworks-repo but can be extended to other repos by just changing the infra. We can change the scoping and the naming in the infra.